src/main/java/moze_intel/projecte/network/commands/ClearKnowledgeCMD.java

```diff
@@ -53,7 +62,20 @@ public void execute(@Nonnull MinecraftServer server, @Nonnull ICommandSender sen
 		}
 	}
 
-	@Override
+    @Override
+    public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos targetPos) {
+        if (args.length == 1)
+        {
+            List<String> list = new ArrayList<>(Arrays.asList(server.getOnlinePlayerNames()));
```

Comment:
Personally I think if you are bothering to implement tab complete you should at least have it be Tab complete instead of tab overwrite like lots of mods implement because of being lazy. For your implementation say there are two people online, bob and alice. If I were to start typing "ali" and hit tab it would show `bob, alice, @p, @a, @r, @e, @s` instead of completing to the one finishing option of "alice". The same goes for the other command's tab complete. In addition if there are no arguments that is when you should be displaying this full list.

This is what I use in my mods for tab complete of player names:
```java
if (args.length == 0) {
    return Arrays.asList(server.getPlayerList().getOnlinePlayerNames());
}
if (args.length == 1) {
    String partialName = args[0];
    return Arrays.stream(server.getPlayerList().getOnlinePlayerNames()).filter(name -> name.startsWith(partialName)).collect(Collectors.toList());
}
```
I would recommend doing something similar (given this does not have the "defaultSelectors" in it).